src/structure/viewer.ts

```diff
@@ -810,6 +797,12 @@ export class MoleculeViewer {
         // ======= color settings
         // setup state when the property changes
         const ColorPropertyChanges = () => {
+            const properties = JSON.parse(JSON.stringify(this._properties)) as Record<
+                string,
+                (number | undefined)[]
```

Comment:
I don't think the values can be undefined here, all the sparsity should be handled by the `environments` list

```suggestion
                number
```